keps/sig-node/4210-max-image-gc-age/README.md

```diff
@@ -0,0 +1,395 @@
+- ImageGCLowThresholdPercent: The lowest disk usage will be before garbage collection begins
+- ImageGCHighThresholdPercent: The highest disk usage will be before garbage collection runs each GC period.
+
+Between each of these options, there is a common thread: image garbage collection only triggers when disk usage has reached a certain threshold.
+In other words, there are no alternative conditions the Kubelet will begin triggering collection. Luckily, this condition is the most important:
+the primary goal of garbage collection is to ensure images don't clutter the disk too much and cause it to fill up needlessly.
+However, having the Kubelet be purely reactive means that images will clutter the disk and cause it to fill up. While there aren't reported cases
+where this causes issues, it is inefficient with disk space and can cause the Kubelet to scramble to save disk space when the threshold is met.
+
+An additional approach is to define a way for an admin to request images are cleaned up after they're unused for a certain period of time.
+This would reduce the frequency of the disk usage hitting the level, and provide an admin more flexibility in how garbage collection is defined.
+
+The proposal of this KEP is to add an option to the KubeletConfiguration object that looks like:
+```
+	// ImageMaximumGCAge is the maximum age an image can be unused before it is garbage collected.
+    // The default of this field is 0, which disables it.
+    // +optional
+	ImageMaximumGCAge metav1.Duration
+```
+
+To begin, this option will be set to 0, which will be interpreted as "disabled". In the future, a more reasonable default may be chosen.
+
+This option will only be adhered to if the feature gate ImageMaximumGCAge is configured for the Kubelet.
+
+### User Stories (Optional)
+
+#### Story 1
+
+- As a cluster admin, I would like my unused images to be garbage collected in a timely manner, and not occupy disk space forever.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+- If set incorrectly the ImageMaximumGCAge option could cause unneeded image pulls. For instance, if a Cron job ran
+once a week, but the ImageMaximumGCAge was set to less than a week, that image would get pulled every week, causing needless
+traffic from the registry
+    - Proper documentation on this is the best way to mitigate this risk.
+    - Defining a good default for this value will be similarly tricky.
+- Reliability of image age
+    - Good testing will mitigate/fix any errors
+- New, undiscovered races
+    - If the max image gc age is set very low, will the kubelet race with itself and remove the image right after pulling it?
+    - May need to define a minimum maximum gc age to prevent races like this.
```

Comment:
I think this is important question to answer on design stage. We can explicitly say that current iteration will set the high threshold.

Reply:
what do you mean by the high threshold @SergeyKanzhelev ?

Reply:
sorry. Let's set the minimul available as 30 minuutes in alpha. And say we can allow less in future